File: CHL-Backend/src/main/java/com/CentralHealthLog/entity/Patient.java
package com.CentralHealthLog.entity;

import lombok.*;

import javax.persistence.*;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String username;

    @Column(nullable = false)
    private String password;

    @Column(nullable = false)
    private String firstname;

    @Column(nullable = false)
    private String lastname;

    private String email;

    @Column(unique = true, nullable = false)
    private String phoneNo;

    private String gender;

    private Integer age;

    private String address;
}

File: CHL-Frontend/src/app/model/patient.ts
export class Patient {
    constructor(
        public id?: number,
        public username?: string|number|null,
        public firstname ?: string|number|null,
        public lastname ?: string|number|null,
        public password?: string|number|null,
        public email ?: string|number|null,
        public phoneNo ?: string|number|null,
        public gender ?: string|number|null,
        public address ?: string|number|null,
        public age ?: string|number|null
    ) { }
}